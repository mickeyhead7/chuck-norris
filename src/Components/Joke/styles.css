.joke {
  background-color: lightgrey;
  margin: 0 0 1rem;
  padding: 1rem;
}
