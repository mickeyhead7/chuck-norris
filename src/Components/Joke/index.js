import React from 'react';
import './styles.css';

const Joke = ({ id, joke }) => (
  <div className="joke">
    {joke}
  </div>
);

export default Joke;
