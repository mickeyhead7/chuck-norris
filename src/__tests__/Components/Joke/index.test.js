import React from 'react';
import { shallow } from 'enzyme';
import Joke from '../../../Components/Joke';

describe('Joke', () => {
  test('Joke default', () => {
    const props = {
      id: 1,
      joke: 'foo'
    };
    const joke = shallow(<Joke {...props} />);

    expect(joke).toMatchSnapshot();
    expect(joke.props().id).toEqual(props.id);
    expect(joke.text()).toEqual(props.joke);
  });
});