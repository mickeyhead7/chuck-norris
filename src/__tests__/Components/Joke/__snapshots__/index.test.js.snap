// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joke Joke default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Joke
    id={1}
    joke="foo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "foo",
      "className": "joke",
      "id": 1,
    },
    "ref": null,
    "rendered": "foo",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "foo",
        "className": "joke",
        "id": 1,
      },
      "ref": null,
      "rendered": "foo",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
